src/test/cpp/CollisionAvoidanceTest.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 FRC Team 3512. All Rights Reserved.
+
+#include <iostream>
+
+#include <frc/Notifier.h>
+#include <frc/simulation/JoystickSim.h>
+#include <frc/simulation/SimHooks.h>
+#include <gtest/gtest.h>
+
+#include "Constants.hpp"
+#include "subsystems/Climber.hpp"
+#include "subsystems/Drivetrain.hpp"
+#include "subsystems/Flywheel.hpp"
+#include "subsystems/Turret.hpp"
+#include "subsystems/Vision.hpp"
+
+namespace {
+class CollisionAvoidanceTest : public testing::Test {
+protected:
+    void SetUp() override { frc::sim::PauseTiming(); }
+
+    void TearDown() override { frc::sim::ResumeTiming(); }
+};
+}  // namespace
+
+TEST_F(CollisionAvoidanceTest, AvoidanceLimits) {
+    frc::sim::JoystickSim appendageStick1{
+        frc3512::Constants::Robot::kAppendageStick1Port};
+
+    frc3512::Vision vision;
+    frc3512::Drivetrain drivetrain;
+    frc3512::Flywheel flywheel{drivetrain};
+    frc3512::Turret turret{vision, drivetrain, flywheel};
+    frc3512::Climber climber{turret};
+
+    frc3512::SubsystemBase::RunAllTeleopInit();
+    frc::Notifier controllerPeriodic{[&] {
+        turret.TeleopPeriodic();
+        turret.ControllerPeriodic();
+        climber.TeleopPeriodic();
+    }};
+    controllerPeriodic.StartPeriodic(frc3512::Constants::kDt);
+
+    appendageStick1.SetTrigger(true);
```

Comment:
```suggestion
    appendageStick1.SetRawButton(1, true);
```
This function is in `JoystickSim`'s `GenericHIDSim` base class.